From 64a2ce52515821c27757e19a44a970ace3bf1c78 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 21 Nov 2025 19:46:16 +0100
Subject: [PATCH] libs: iksemel: fix compilation error for iks_stack_delete

Fix compilation error

hash.c: In function 'hash_delete':
hash.c:141:28: error: passing argument 1 of 'iks_stack_delete' from incompatible pointer type [-Wincompatible-pointer-types]
  141 |         iks_stack_delete (h->s);
      |                           ~^~~
      |                            |
      |                            ikstack * {aka struct ikstack_struct *}
In file included from hash.c:8:
../include/iksemel.h:27:34: note: expected 'ikstack **' {aka 'struct ikstack_struct **'} but argument is of type 'ikstack *' {aka 'struct ikstack_struct *'}
   27 | void iks_stack_delete (ikstack **sp);
      |                        ~~~~~~~~~~^~

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 libs/iksemel/tools/hash.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libs/iksemel/tools/hash.c
+++ b/libs/iksemel/tools/hash.c
@@ -138,7 +138,7 @@ hash_print (hash *h, char *title_fmt, ch
 void
 hash_delete (hash *h)
 {
-	iks_stack_delete (h->s);
+	iks_stack_delete (&h->s);
 	free (h->table);
 	free (h);
 }
